proposals/NNNN-GetSystemTime.md

```diff
@@ -0,0 +1,63 @@
+# GetSystemTime RPC
+
+* Proposal: [SDL-NNNN](NNNN-GetSystemTime.md)
+* Author: [Aleksandr Stasiuk](https://github.com/AStasiuk)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core / RPC]
+
+## Introduction
+
+This proposal is about adding a new RPC GetSystemTime to SDL Core for get real system time during security TLS handshake.
+
+
+## Motivation
+
+In some cases SDL Core might not have access to the real system time so needed an interface to the system in order to request it when it is required. System time is necessary for security certificates validation and successful finishing security TLS handshake in case using of Mobile Navigation application on mobile phone.
+
+
+## Proposed solution
+
+
+The solution proposes adding new RPC between SDL and HMI.  GetSystemTime gives possibility to get the real system time status by introducing new API.
+
+SDL Core must send this GetSystemTime RPC to HMI for obtaining system time in case Security TLS Handshake.
+
+Expected that HMI will provide real system time with GetSystemTime response.
+
+## Additions to HMI_API
+
+GetSystemTime request\response
+
+```xml
+<interface name="BasicCommunication" version="< >" date="< >">
+<function name="GetSystemTime" messagetype="request">
+    <description>Request from SDL to HMI to obtain current UTC time.</description>
+</function>
+<function name="GetSystemTime" messagetype="response">
+    <param name="systemTime" type="Common.DateTime" mandatory="true">
+      <description>Current UTC system time</description>
+    </param>
+  </function>
+```
+OnSystemTimeReady notification
+
+```xml
+<interface name="BasicCommunication" version="< >" date="< >">
+<function name="OnSystemTimeReady" messagetype="notification">
+    <description>HMI must notify SDL about readiness to provide system time.</description>
+</function>
+```
+
+## Potential downsides
+
+N/A
+
+## Impact on existing code
+
+Requires changes on SDL core,  and HMI_API.
+
+Would require a major version change.

```

Comment:
We do not think this requires a major version change - especially if SDL Core can use the existing platform time if HMI fails to respond to this RPC within the timeout period.
In fact if HMI does not send OnSystemTimeReady, then SDL Core should not send the GetSystemTime request.

Reply:
@robinmk Fixed in https://github.com/smartdevicelink/sdl_evolution/pull/324/commits/19088cdb433d915b8fe826acfcf622b427a02f1e